Index: chinachu/chinachu
===================================================================
--- chinachu.orig/chinachu	2014-06-07 20:27:08.683045171 +0900
+++ chinachu/chinachu	2014-06-07 20:33:24.468908594 +0900
@@ -16,7 +16,7 @@
 shopt -s extglob
 
 # variables
-NODE_VER=0.10.26
+NODE_VER=`/usr/bin/nodejs --version | sed 's/^v//'`
 
 main () {
   local SELF_PATH DIR SYM
@@ -40,10 +40,10 @@
   done
   
   # path
-  CHINACHU_DIR=$(dirname -- "$SELF_PATH")
+  CHINACHU_DIR=/var/lib/chinachu
   NAVE_DIR=${CHINACHU_DIR}/.nave
-  NODE_PATH=${NAVE_DIR}/installed/${NODE_VER}/bin/node
-  NPM_PATH=${NAVE_DIR}/installed/${NODE_VER}/bin/npm
+  NODE_PATH=${CHINACHU_DIR}/node_modules/
+  NPM_PATH=/usr/bin/npm
   USR_DIR=${CHINACHU_DIR}/usr
   
   export PATH=$USR_DIR/bin:${NAVE_DIR}/installed/${NODE_VER}/bin:$PATH
