Index: chinachu/chinachu
===================================================================
--- chinachu.orig/chinachu	2014-04-05 14:26:54.964365289 +0900
+++ chinachu/chinachu	2014-04-05 14:26:54.956365249 +0900
@@ -64,13 +64,17 @@
   local cmd="$1"
   shift
   case $cmd in
-    installer | updater | service | update | search | \
+    service | update | search | \
     reserve | unreserve | skip | unskip | stop | \
     rule | enrule | disrule | rmrule | \
     rules | reserves | recording | recorded | \
-    cleanup | unlock | ircbot | test )
+    cleanup | unlock | ircbot )
       cmd="chinachu_$cmd"
       ;;
+    installer | updater | test )
+      echo "Disable $cmd by debian package."
+      return 0;
+      ;;
     * )
       cmd="chinachu_help"
       ;;
