Index: chinachu/app-cli.js
===================================================================
--- chinachu.orig/app-cli.js	2014-06-07 20:55:03.927352263 +0900
+++ chinachu/app-cli.js	2014-06-07 20:55:03.915352200 +0900
@@ -5,12 +5,12 @@
  *  http://chinachu.akkar.in/
 **/
 
-var CONFIG_FILE         = __dirname + '/config.json';
-var RULES_FILE          = __dirname + '/rules.json';
-var RESERVES_DATA_FILE  = __dirname + '/data/reserves.json';
-var SCHEDULE_DATA_FILE  = __dirname + '/data/schedule.json';
-var RECORDING_DATA_FILE = __dirname + '/data/recording.json';
-var RECORDED_DATA_FILE  = __dirname + '/data/recorded.json';
+var CONFIG_FILE         = '/etc/chinachu/config.json';
+var RULES_FILE          = '/var/lib/chinachu/rules.json';
+var RESERVES_DATA_FILE  = '/var/lib/chinachu/data/reserves.json';
+var SCHEDULE_DATA_FILE  = '/var/lib/chinachu/data/schedule.json';
+var RECORDING_DATA_FILE = '/var/lib/chinachu/data/recording.json';
+var RECORDED_DATA_FILE  = '/var/lib/chinachu/data/recorded.json';
 
 // 標準モジュールのロード
 var fs            = require('fs');
Index: chinachu/app-operator.js
===================================================================
--- chinachu.orig/app-operator.js	2014-06-07 20:55:03.927352263 +0900
+++ chinachu/app-operator.js	2014-06-07 20:55:03.919352223 +0900
@@ -8,10 +8,10 @@
 /*global gc */
 'use strict';
 
-var CONFIG_FILE         = __dirname + '/config.json';
-var RESERVES_DATA_FILE  = __dirname + '/data/reserves.json';
-var RECORDING_DATA_FILE = __dirname + '/data/recording.json';
-var RECORDED_DATA_FILE  = __dirname + '/data/recorded.json';
+var CONFIG_FILE         = '/etc/chinachu/config.json';
+var RESERVES_DATA_FILE  = '/var/lib/chinachu/data/reserves.json';
+var RECORDING_DATA_FILE = '/var/lib/chinachu/data/recording.json';
+var RECORDED_DATA_FILE  = '/var/lib/chinachu/data/recorded.json';
 
 // 標準モジュールのロード
 var path          = require('path');
Index: chinachu/app-wui.js
===================================================================
--- chinachu.orig/app-wui.js	2014-06-07 20:55:03.927352263 +0900
+++ chinachu/app-wui.js	2014-06-07 20:55:03.919352223 +0900
@@ -8,15 +8,15 @@
 /*global gc */
 'use strict';
 
-var CONFIG_FILE         = __dirname + '/config.json';
-var RULES_FILE          = __dirname + '/rules.json';
-var RESERVES_DATA_FILE  = __dirname + '/data/reserves.json';
-var SCHEDULE_DATA_FILE  = __dirname + '/data/schedule.json';
-var RECORDING_DATA_FILE = __dirname + '/data/recording.json';
-var RECORDED_DATA_FILE  = __dirname + '/data/recorded.json';
-var OPERATOR_LOG_FILE   = __dirname + '/log/operator';
-var WUI_LOG_FILE        = __dirname + '/log/wui';
-var SCHEDULER_LOG_FILE  = __dirname + '/log/scheduler';
+var CONFIG_FILE         = '/etc/chinachu/config.json';
+var RULES_FILE          = '/var/lib/chinachu/rules.json';
+var RESERVES_DATA_FILE  = '/var/lib/chinachu/data/reserves.json';
+var SCHEDULE_DATA_FILE  = '/var/lib/chinachu/data/schedule.json';
+var RECORDING_DATA_FILE = '/var/lib/chinachu/data/recording.json';
+var RECORDED_DATA_FILE  = '/var/lib/chinachu/data/recorded.json';
+var OPERATOR_LOG_FILE   = '/var/log/chinachu/operator';
+var WUI_LOG_FILE        = '/var/log/chinachu/wui';
+var SCHEDULER_LOG_FILE  = '/var/log/chinachu/scheduler';
 
 // Load Config
 var config = require(CONFIG_FILE);
@@ -852,8 +852,8 @@
 	
 	var c = chinachu.createCountdown(2, chinachu.createTimeout(processChecker, 5000));
 	
-	if (fs.existsSync('/var/run/chinachu-operator.pid') === true) {
-		fs.readFile('/var/run/chinachu-operator.pid', function (err, pid) {
+	if (fs.existsSync('/var/run/chinachu/chinachu-operator.pid') === true) {
+		fs.readFile('/var/run/chinachu/chinachu-operator.pid', function (err, pid) {
 			
 			if (err) { return c.tick(); }
 			
@@ -881,8 +881,8 @@
 		c.tick();
 	}
 	
-	if (fs.existsSync('/var/run/chinachu-wui.pid') === true) {
-		fs.readFile('/var/run/chinachu-wui.pid', function (err, pid) {
+	if (fs.existsSync('/var/run/chinachu/chinachu-wui.pid') === true) {
+		fs.readFile('/var/run/chinachu/chinachu-wui.pid', function (err, pid) {
 			
 			if (err) { return c.tick(); }
 			
Index: chinachu/app-scheduler.js
===================================================================
--- chinachu.orig/app-scheduler.js	2014-06-07 20:55:03.927352263 +0900
+++ chinachu/app-scheduler.js	2014-06-07 20:55:03.919352223 +0900
@@ -8,12 +8,12 @@
 /*global gc */
 'use strict';
 
-var PID_FILE = __dirname + '/data/scheduler.pid';
+var PID_FILE ='/var/run/chinachu/chinachu-scheduler.pid';
 
-var CONFIG_FILE         = __dirname + '/config.json';
-var RULES_FILE          = __dirname + '/rules.json';
-var RESERVES_DATA_FILE  = __dirname + '/data/reserves.json';
-var SCHEDULE_DATA_FILE  = __dirname + '/data/schedule.json';
+var CONFIG_FILE         = '/etc/chinachu/config.json';
+var RULES_FILE          = '/var/lib/chinachu/rules.json';
+var RESERVES_DATA_FILE  = '/var/lib/chinachu/data/reserves.json';
+var SCHEDULE_DATA_FILE  = '/var/lib/chinachu/data/schedule.json';
 
 // 標準モジュールのロード
 var path          = require('path');
Index: chinachu/chinachu
===================================================================
--- chinachu.orig/chinachu	2014-06-07 20:55:03.927352263 +0900
+++ chinachu/chinachu	2014-06-07 20:55:03.919352223 +0900
@@ -16,7 +16,7 @@
 shopt -s extglob
 
 # variables
-NODE_VER=0.10.26
+NODE_VER=`/usr/bin/nodejs --version | sed 's/^v//'`
 
 main () {
   local SELF_PATH DIR SYM
@@ -40,10 +40,10 @@
   done
   
   # path
-  CHINACHU_DIR=$(dirname -- "$SELF_PATH")
+  CHINACHU_DIR=/var/lib/chinachu
   NAVE_DIR=${CHINACHU_DIR}/.nave
-  NODE_PATH=${NAVE_DIR}/installed/${NODE_VER}/bin/node
-  NPM_PATH=${NAVE_DIR}/installed/${NODE_VER}/bin/npm
+  NODE_PATH=${CHINACHU_DIR}/node_modules/
+  NPM_PATH=/usr/bin/npm
   USR_DIR=${CHINACHU_DIR}/usr
   
   export PATH=$USR_DIR/bin:${NAVE_DIR}/installed/${NODE_VER}/bin:$PATH
