use system python binary.
--- a/web/lib/ace/kitchen-sink/docs/python.py
+++ b/web/lib/ace/kitchen-sink/docs/python.py
@@ -1,4 +1,4 @@
-#!/usr/local/bin/python
+#!/usr/bin/python
 
 import string, sys
 
--- a/web/lib/ace/kitchen-sink/docs/coffee.coffee
+++ b/web/lib/ace/kitchen-sink/docs/coffee.coffee
@@ -1,4 +1,4 @@
-#!/usr/bin/env coffee
+#!/usr/bin/coffee
 
 try
     throw URIError decodeURI(0xC0ffee * 123456.7e-8 / .9)
--- a/web/lib/ace/kitchen-sink/docs/coffeescript.coffee
+++ b/web/lib/ace/kitchen-sink/docs/coffeescript.coffee
@@ -1,4 +1,4 @@
-#!/usr/bin/env coffee
+#!/usr/bin/coffee
 
 try
     throw URIError decodeURI(0xC0ffee * 123456.7e-8 / .9)
